### PR TITLE
br-stream: add region context in applyRequest

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31975

Problem Summary:
apply kv files to tikv cluster.

### What is changed and how it works?
use raft router to apply kv files instead of kv engine.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
see #https://github.com/3pointer/tikv/pull/24/files
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/restore/split.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func CheckRegionConsistency(startKey, endKey []byte, regions []*RegionInfo) erro
 func PaginateScanRegion(
 	ctx context.Context, client SplitClient, startKey, endKey []byte, limit int,
 ) ([]*RegionInfo, error) {
-	if len(endKey) != 0 && bytes.Compare(startKey, endKey) >= 0 {
+	if len(endKey) != 0 && bytes.Compare(startKey, endKey) > 0 {
 		return nil, errors.Annotatef(berrors.ErrRestoreInvalidRange, "startKey >= endKey, startKey: %s, endkey: %s",
```
Comment (joccau) [inline br/pkg/restore/split.go:435]: Update log corresponding to code
```suggestion
		return nil, errors.Annotatef(berrors.ErrRestoreInvalidRange, "startKey > endKey, startKey: %s, endkey: %s",
```

**Review by joccau — APPROVED** [review]
LGTM for rest

**Pull request merged**